app/lib/project/MetapolatorProject.js

```diff
@@ -509,8 +509,54 @@ define([
         return this._controller;
     };
 
-    _p.import = function(masterName, sourceUFODir, glyphs) {
-        var importer = new ImportController( this._log, this,
+    _p.importZippedUFOInstances = function(filename, blob) {
+        var mem_io = new InMemory()
+          , imported_instances = Array()
+          ;
+
+        zipUtil.unpack(false, blob, mem_io, "");
+
+        var dirs = mem_io.readDir(false, "/")
+          , baseDir = dirs[0]
+          , names = mem_io.readDir(false, baseDir)
+          , n, l
+          ;
+
+        console.log("dirs:", dirs);
+
+        for (n=0, l=names.length; n<l; n++){
+            var name = names[n]
+              , UFOZip = baseDir + name //This may be wrong in some cases. We need a more robust implementation.
```

Comment:
a good start for more robust implementation would  be to split this implementation into smaller bites i.e. several functions, that perform specific tasks. The not robust function could then be replaced by a more robust one.

As I understood it, there are different schemes how a ufo could be packed into a zip. Maybe we should define the two or three acceptable versions and try to detect them and then dispatch the differing part of importing them to different methods.

The three posible methods I have in mind are:

* myfont.ufoz <- zipped at base dir level, fontinfo.plist etc. is directly in the first level
* myfont.ufo.zip/myfont.ufoz
* myfont.ufo.zip/myfont.ufo/fontinfo.plist

Which are the ones that we offer for download?

Which is the one a user is most likely to produce when saying *"zip this ufo-directory into a zip file"*